sys/net/gnrc/sock/gnrc_sock.c

```diff
@@ -89,7 +89,8 @@ void gnrc_sock_create(gnrc_sock_reg_t *reg, gnrc_nettype_t type, uint32_t demux_
 }
 
 ssize_t gnrc_sock_recv(gnrc_sock_reg_t *reg, gnrc_pktsnip_t **pkt_out,
-                       uint32_t timeout, sock_ip_ep_t *remote)
+                       uint32_t timeout, sock_ip_ep_t *remote,
+                       sock_ip_ep_t *local)
```

Comment:
I think it makes sense to introduce a `gnrc_sock_rx_aux_t` structure here instead of using `sock_ip_ep_t *local`. In the end, every additional member in `sock_udp_aux_rx_t` will be populated by this function. Maybe we don't want to increase the number of arguments for every member, especially since most applications will only use a small or even empty subset of the information obtainable via aux.

Reply:
Agreed. Since I was wondering myself, but than also found the answer myself: The `sock_udp_aux_rx_t` can not just duplicated here, as every sock type has their own `sock_*_aux_rx_t`.